Library/Homebrew/os/linux/ld.rb

```diff
@@ -0,0 +1,75 @@
+# typed: true
+# frozen_string_literal: true
+
+module OS
+  module Linux
+    # Helper functions for querying `ld` information.
+    #
+    # @api private
+    module Ld
+      module_function
+
+      sig { returns(String) }
+      def brewed_ld_so_diagnostics
```

Comment:
Would be nice to have some unit tests for this module maybe?

Reply:
Added a unit test particularly for `library_paths` which is parsing the `ld.so.conf` files (IMO, the least trivial part of the module).

It made more sense to provide one big input (containing multiple corner cases) for one test case and then confirm that all the expected corner cases are reflected in the output - let me know if you have a better/more idiomatic approach in mind.